### PR TITLE
feat(bridge): [ISSUE-333] Bridge: Integrate cargo-spellcheck

### PR DESCRIPTION
**Pull request by TomaszWaszczyk**
I test how to contribute to the repository (that is why `Draft pull request`), when the PR will be ready to merge I will mark it as vanilla plain PR :D

### CODE REVIEW

File: .config/spellcheck.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# search_dirs = []
+
+# Adds additional dictionaries, can be specified as
+# absolute paths or relative in the search dirs (in this order).
+# Relative paths are resolved relative to the configuration file
+# which is used.
+# Refer to `man 5 hunspell`
+# or https://www.systutorials.com/docs/linux/man/4-hunspell/#lbAE
+# on how to define a custom dictionary file.
+extra_dictionaries = []
```
Comment (tomusdrw) [inline .config/spellcheck.toml:30]: I thought that we will need a custom dictionary to make the check work?

**Review by tomusdrw — COMMENTED** [review]
Looks like a good start, would be good to get that integrated to the CI pipeline as well :)

**Comment by HCastano** [comment]
Thanks for your PR! Note that there's no need to have the issue name in the PR title. It would actually be better to include it as part of the PR description using GitHub [keywords](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords), like so: "Closes #ISSUE_NUMBER". That'll automatically automatically close the issue when this PR gets merged.

**Comment by tomusdrw** [comment]
@TomaszWaszczyk don't hesitate to get in touch with us on Element if you got stuck, happy to help driving this PR to completion.

**Comment by TomaszWaszczyk** [comment]
Will finish it on weekend, currently I have high tension with on-daily job. Thanks for the ping. ;-)

**Comment by TomaszWaszczyk** [comment]
I was trying to add new dependency to `Cargo.toml` but when I want to do this, I got:

image

Do I good think that in order to integrate `cargo-spellcheck` I need firstly to add the dependency to `Cargo.toml`?

**Comment by HCastano** [comment]
> I was trying to add new dependency to `Cargo.toml` but when I want to do this, I got:
> 
> image
> 
> Do I good think that in order to integrate `cargo-spellcheck` I need firstly to add the dependency to `Cargo.toml`?

Hey. I don't think you should be adding `cargo spellcheck` as a dependency in the project. The goal here is to install it on the CI server (the GH Actions runners), and then run it from there. The only things we need to add to the repo are `cargo spellcheck`'s configuration files

**Comment by HCastano** [comment]
@TomaszWaszczyk Hey, do you need any help with this?

File: .config/spellcheck.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+lang = "en_US"
+# OS specific additives
+# Linux: [ /usr/share/myspell ]
+# Windows: []
+# macOS [ /home/alice/Libraries/hunspell, /Libraries/hunspell ]
+
+# Additional search paths, which take presedence over the default
+# os specific search dirs, searched in order, defaults last
+search_dirs = ["./hunspell/"]
+
```
Comment (TomaszWaszczyk) [inline .config/spellcheck.toml:22]: @HCastano I have created the file, what is wrong that CI dont want to find it?

**Comment by HCastano** [comment]
So I think first what we need to do is install `hunspell` on the CI machine. You should be able to add it with `sudo apt install hunspell` somewhere in the `lint.yml` file. That should give us a default dictionary to work with. I _think_ that'll get rid of the error you're getting, since it doesn't look like it actually has to do with the custom dictionary.

After that I suggest you move the custom dictionary you've defined to `.config/lingo` so it doesn't clash with what `hunspell` installs.

**Comment by tomusdrw** [comment]
@TomaszWaszczyk closing for now, but I would be very happy if you found time to finish this. Feel free to re open any time.

**Pull request closed without merging**